keras_cv/models/convnext.py

```diff
@@ -0,0 +1,494 @@
+        [0, 1].
+      projection_dim (int): Projection dimensionality.
+
+    Returns:
+      Tensor multiplied to the scale.
+    """
+
+    def __init__(self, init_values, projection_dim, **kwargs):
+        super().__init__(**kwargs)
+        self.init_values = init_values
+        self.projection_dim = projection_dim
+
+    def build(self, input_shape):
+        self.gamma = tf.Variable(self.init_values * tf.ones((self.projection_dim,)))
+
+    def call(self, x):
+        return x * self.gamma
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "init_values": self.init_values,
+                "projection_dim": self.projection_dim,
+            }
+        )
+        return config
+
+
+def ConvNeXtBlock(
+    projection_dim, drop_path_rate=0.0, layer_scale_init_value=1e-6, name=None
+):
+    """ConvNeXt block.
+
+    References:
+      - https://arxiv.org/abs/2201.03545
+      - https://github.com/facebookresearch/ConvNeXt/blob/main/models/convnext.py
+
+    Notes:
+      In the original ConvNeXt implementation (linked above), the authors use
+      `Dense` layers for pointwise convolutions for increased efficiency.
+      Following that, this implementation also uses the same.
+
+    Args:
+      projection_dim (int): Number of filters for convolution layers. In the
+        ConvNeXt paper, this is referred to as projection dimension.
+      drop_path_rate (float): Probability of dropping paths. Should be within
+        [0, 1].
+      layer_scale_init_value (float): Layer scale value. Should be a small float
+        number.
+      name: name to path to the keras layer.
+
+    Returns:
+      A function representing a ConvNeXtBlock block.
+    """
+    if name is None:
+        name = "prestem" + str(backend.get_uid("prestem"))
+
+    def apply(inputs):
+        x = inputs
+
+        x = layers.Conv2D(
+            filters=projection_dim,
+            kernel_size=7,
+            padding="same",
+            groups=projection_dim,
+            name=name + "_depthwise_conv",
+        )(x)
+        x = layers.LayerNormalization(epsilon=1e-6, name=name + "_layernorm")(x)
+        x = layers.Dense(4 * projection_dim, name=name + "_pointwise_conv_1")(x)
+        x = layers.Activation("gelu", name=name + "_gelu")(x)
+        x = layers.Dense(projection_dim, name=name + "_pointwise_conv_2")(x)
+
+        if layer_scale_init_value is not None:
+            x = LayerScale(
+                layer_scale_init_value,
+                projection_dim,
+                name=name + "_layer_scale",
+            )(x)
+        if drop_path_rate:
+            layer = StochasticDepth(drop_path_rate, name=name + "_stochastic_depth")
+            return layer([inputs, x])
+        else:
+            layer = layers.Activation("linear", name=name + "_identity")
+            return inputs + layer(x)
+
+    return apply
+
+
+def Head(num_classes=1000, activation="softmax", name=None):
```

Comment:
Let's not include a default here -- this should be a required param, as we don't want to implicitly expect ImageNet